docs/design/2022-10-13-mysql-compatible-auto-increment.md

```diff
@@ -0,0 +1,118 @@
+# Proposal: MySQL compatible `AUTO_INCREMENT`
+
+- Author(s): [tiancaiamao](https://github.com/tiancaiamao)
+- Discussion PR: https://github.com/pingcap/tidb/pull/38449
+- Tracking Issue: https://github.com/pingcap/tidb/issues/38442
+
+## Abstract
+
+This proposes an implementation of making the `AUTO_INCREMENT` behaviour compatible with MySQL.
+
+## Background
+
+MySQL deploys on a single-machine, and it provides the [`AUTO_INCREMENT`](https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html) table attribute to generate a unique identity for new rows.
+
+In TiDB, we support generating `AUTO_INCREMENT` IDs that are **unique**, **monotone increasing**, but the IDs may **not be consecutive**.
+
+The current behaviour in TiDB is not fully-compatible with MySQL.
+
+TiDB is a distributed database, each TiDB instance caches a batch of IDs for local allocating. When it exhaust its local cached IDs, it ask for another batch. For example, TiDB instance A may get ID range [0-20000), and instance B takes away range [20000, 40000), the next batch for TiDB instance A could be [40000, 60000), that's why the IDs are not **consecutive**. The ID sequence from instance A might be ...19998, 19999, [a hole here], 40000, 40001...
+
+## Proposal
+
+This proposes that we introduce a centralized auto ID allocating service, the `AUTO_INCREMENT` ID is allocated from the centralized service and there is no caching mechanism on the TiDB layer, so the behaviour can be MySQL compatible.
+
+## Compatibility
```

Comment:
```suggestion
## Compatibility
```
IMHO a better title for this is 'syntax', we are actually talking about the 'syntax compatibility'.